v3.1 build fails on macOS

```
:info:build make: Entering directory `/opt/local/var/macports/build/_Users_Admin_Developer_Ports_fuse_btfs/btfs/work/btfs-3.1'
:info:build Making all in src
:info:build make[1]: Entering directory `/opt/local/var/macports/build/_Users_Admin_Developer_Ports_fuse_btfs/btfs/work/btfs-3.1/src'
:info:build /usr/bin/clang++ -DPACKAGE_NAME=\"btfs\" -DPACKAGE_TARNAME=\"btfs\" -DPACKAGE_VERSION=\"3.1\" -DPACKAGE_STRING=\"btfs\ 3.1\" -DPACKAGE_BUGREPORT=\"johan.gunnarsson@gmail.com\" -DPACKAGE_URL=\"https://github.com/johang/btfs\" -DPACKAGE=\"btfs\" -DVERSION=\"3.1\" -DHAVE_STDIO_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1 -DHAVE_STRINGS_H=1 -DHAVE_SYS_STAT_H=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_UNISTD_H=1 -DSTDC_HEADERS=1 -DHAVE_MEMSET=1 -DHAVE_MEMCPY=1 -DHAVE_STRCMP=1 -DHAVE_MKDIR=1 -DHAVE_STRDUP=1 -DHAVE_REALPATH=1 -DHAVE_GETXATTR=1 -DHAVE_DIRNAME=1 -DHAVE_BASENAME=1 -DHAVE_LIBPTHREAD=1 -I.   -I/opt/local/include -I/opt/local/include/fuse -isysroot/Library/Developer/CommandLineTools/SDKs/MacOSX15.sdk -std=c++14 -Wall -Wextra -Wconversion -Wsign-compare -Wsign-conversion -Wno-unused-parameter -I/opt/local/include/fuse3 -fexceptions -DTORRENT_LINKING_SHARED -DBOOST_ASIO_ENABLE_CANCELIO -DBOOST_ASIO_NO_DEPRECATED -DTORRENT_DISABLE_LOGGING -DTORRENT_USE_OPENSSL -DTORRENT_USE_LIBCRYPTO -DTORRENT_SSL_PEERS -DOPENSSL_NO_SSL2 -I/opt/local/libexec/openssl3/include -I/opt/local/libexec/openssl3/include -pipe -Os -stdlib=libc++ -isysroot/Library/Developer/CommandLineTools/SDKs/MacOSX15.sdk -arch arm64 -MT btfs-btfs.o -MD -MP -MF .deps/btfs-btfs.Tpo -c -o btfs-btfs.o `test -f 'btfs.cc' || echo './'`btfs.cc
:info:build /usr/bin/clang++ -DPACKAGE_NAME=\"btfs\" -DPACKAGE_TARNAME=\"btfs\" -DPACKAGE_VERSION=\"3.1\" -DPACKAGE_STRING=\"btfs\ 3.1\" -DPACKAGE_BUGREPORT=\"johan.gunnarsson@gmail.com\" -DPACKAGE_URL=\"https://github.com/johang/btfs\" -DPACKAGE=\"btfs\" -DVERSION=\"3.1\" -DHAVE_STDIO_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1 -DHAVE_STRINGS_H=1 -DHAVE_SYS_STAT_H=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_UNISTD_H=1 -DSTDC_HEADERS=1 -DHAVE_MEMSET=1 -DHAVE_MEMCPY=1 -DHAVE_STRCMP=1 -DHAVE_MKDIR=1 -DHAVE_STRDUP=1 -DHAVE_REALPATH=1 -DHAVE_GETXATTR=1 -DHAVE_DIRNAME=1 -DHAVE_BASENAME=1 -DHAVE_LIBPTHREAD=1 -I.   -I/opt/local/include -I/opt/local/include/fuse -isysroot/Library/Developer/CommandLineTools/SDKs/MacOSX15.sdk -std=c++14 -Wall -Wextra -Wconversion -Wsign-compare -Wsign-conversion -Wno-unused-parameter -pipe -Os -stdlib=libc++ -isysroot/Library/Developer/CommandLineTools/SDKs/MacOSX15.sdk -arch arm64 -MT btfsstat-btfsstat.o -MD -MP -MF .deps/btfsstat-btfsstat.Tpo -c -o btfsstat-btfsstat.o `test -f 'btfsstat.cc' || echo './'`btfsstat.cc
:info:build In file included from btfs.cc:30:
:info:build /opt/local/include/fuse3/fuse.h:393:2: error: a type specifier is required for all declarations
:info:build   393 |         FUSE_DARWIN_EXTEND_OPERATION(
:info:build       |         ^
:info:build /opt/local/include/fuse3/fuse_common.h:77:2: note: expanded from macro 'FUSE_DARWIN_EXTEND_OPERATION'
:info:build    77 |         typeof(darwin) name;
:info:build       |         ^
:info:build In file included from btfs.cc:30:
:info:build /opt/local/include/fuse3/fuse.h:394:3: error: expected ';' at end of declaration list
:info:build   394 |                 getattr,
:info:build       |                 ^
:info:build /opt/local/include/fuse3/fuse.h:577:2: error: a type specifier is required for all declarations
:info:build   577 |         FUSE_DARWIN_EXTEND_OPERATION(
:info:build       |         ^
:info:build /opt/local/include/fuse3/fuse_common.h:77:2: note: expanded from macro 'FUSE_DARWIN_EXTEND_OPERATION'
:info:build    77 |         typeof(darwin) name;
:info:build       |         ^
:info:build In file included from btfs.cc:30:
:info:build /opt/local/include/fuse3/fuse.h:578:3: error: expected ';' at end of declaration list
:info:build   578 |                 statfs,
:info:build       |                 ^
:info:build /opt/local/include/fuse3/fuse.h:635:2: error: a type specifier is required for all declarations
:info:build   635 |         FUSE_DARWIN_EXTEND_OPERATION(
:info:build       |         ^
:info:build /opt/local/include/fuse3/fuse_common.h:77:2: note: expanded from macro 'FUSE_DARWIN_EXTEND_OPERATION'
:info:build    77 |         typeof(darwin) name;
:info:build       |         ^
:info:build In file included from btfs.cc:30:
:info:build /opt/local/include/fuse3/fuse.h:636:3: error: expected ';' at end of declaration list
:info:build   636 |                 setxattr,
:info:build       |                 ^
:info:build /opt/local/include/fuse3/fuse.h:643:2: error: a type specifier is required for all declarations
:info:build   643 |         FUSE_DARWIN_EXTEND_OPERATION(
:info:build       |         ^
:info:build /opt/local/include/fuse3/fuse_common.h:77:2: note: expanded from macro 'FUSE_DARWIN_EXTEND_OPERATION'
:info:build    77 |         typeof(darwin) name;
:info:build       |         ^
:info:build In file included from btfs.cc:30:
:info:build /opt/local/include/fuse3/fuse.h:644:3: error: expected ';' at end of declaration list
:info:build   644 |                 getxattr,
:info:build       |                 ^
:info:build /opt/local/include/fuse3/fuse.h:687:1: error: a type specifier is required for all declarations
:info:build   687 | FUSE_DARWIN_EXTEND_OPERATION(
:info:build       | ^
:info:build /opt/local/include/fuse3/fuse_common.h:77:2: note: expanded from macro 'FUSE_DARWIN_EXTEND_OPERATION'
:info:build    77 |         typeof(darwin) name;
:info:build       |         ^
:info:build In file included from btfs.cc:30:
:info:build /opt/local/include/fuse3/fuse.h:688:2: error: expected ';' at end of declaration list
:info:build   688 |         readdir,
:info:build       |         ^
:info:build /opt/local/include/fuse3/fuse.h:1385:2: error: expected function body after function declarator
:info:build  1385 |         fuse_fs_getattr,
:info:build       |         ^
:info:build /opt/local/include/fuse3/fuse.h:1422:2: error: expected function body after function declarator
:info:build  1422 |         fuse_fs_statfs,
:info:build       |         ^
:info:build /opt/local/include/fuse3/fuse.h:1429:2: error: expected function body after function declarator
:info:build  1429 |         fuse_fs_readdir,
:info:build       |         ^
:info:build /opt/local/include/fuse3/fuse.h:1462:2: error: expected function body after function declarator
:info:build  1462 |         fuse_fs_setxattr,
:info:build       |         ^
:info:build /opt/local/include/fuse3/fuse.h:1469:2: error: expected function body after function declarator
:info:build  1469 |         fuse_fs_getxattr,
:info:build       |         ^
:info:build In file included from btfs.cc:39:
:info:build /opt/local/include/libtorrent/torrent_info.hpp:50:10: fatal error: 'boost/shared_array.hpp' file not found
:info:build    50 | #include <boost/shared_array.hpp>
:info:build       |          ^~~~~~~~~~~~~~~~~~~~~~~~
:info:build mv -f .deps/btfsstat-btfsstat.Tpo .deps/btfsstat-btfsstat.Po
:info:build /usr/bin/clang++ -std=c++14 -Wall -Wextra -Wconversion -Wsign-compare -Wsign-conversion -Wno-unused-parameter -pipe -Os -stdlib=libc++ -isysroot/Library/Developer/CommandLineTools/SDKs/MacOSX15.sdk -arch arm64  -L/opt/local/lib -Wl,-headerpad_max_install_names -Wl,-syslibroot,/Library/Developer/CommandLineTools/SDKs/MacOSX15.sdk -arch arm64 -o btfsstat btfsstat-btfsstat.o  -lpthread 
:info:build 16 errors generated.
```
This issue is also occurring with `archivemount`, another fuse fs. Reverting https://github.com/johang/btfs/commit/e0659b09a9e4f8c5f04382e33d0c2940940bb9dd and build using fuse 2 will resolve this.

If I set `-std=gnu++14` in Makefile.am, I get a different set of errors:
```
:info:build make: Entering directory `/opt/local/var/macports/build/_Users_Admin_Developer_Ports_fuse_btfs/btfs/work/btfs-3.1'
:info:build Making all in src
:info:build make[1]: Entering directory `/opt/local/var/macports/build/_Users_Admin_Developer_Ports_fuse_btfs/btfs/work/btfs-3.1/src'
:info:build /usr/bin/clang++ -DPACKAGE_NAME=\"btfs\" -DPACKAGE_TARNAME=\"btfs\" -DPACKAGE_VERSION=\"3.1\" -DPACKAGE_STRING=\"btfs\ 3.1\" -DPACKAGE_BUGREPORT=\"johan.gunnarsson@gmail.com\" -DPACKAGE_URL=\"https://github.com/johang/btfs\" -DPACKAGE=\"btfs\" -DVERSION=\"3.1\" -DHAVE_STDIO_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1 -DHAVE_STRINGS_H=1 -DHAVE_SYS_STAT_H=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_UNISTD_H=1 -DSTDC_HEADERS=1 -DHAVE_MEMSET=1 -DHAVE_MEMCPY=1 -DHAVE_STRCMP=1 -DHAVE_MKDIR=1 -DHAVE_STRDUP=1 -DHAVE_REALPATH=1 -DHAVE_GETXATTR=1 -DHAVE_DIRNAME=1 -DHAVE_BASENAME=1 -DHAVE_LIBPTHREAD=1 -I.   -I/opt/local/libexec/openssl3/include -I/opt/local/libexec/boost/1.81/include -isystem/opt/local/include -I/opt/local/include/fuse -isysroot/Library/Developer/CommandLineTools/SDKs/MacOSX15.sdk -std=gnu++14 -Wall -Wextra -Wconversion -Wsign-compare -Wsign-conversion -Wno-unused-parameter -I/opt/local/include/fuse3 -fexceptions -DTORRENT_LINKING_SHARED -DBOOST_ASIO_ENABLE_CANCELIO -DBOOST_ASIO_NO_DEPRECATED -DTORRENT_DISABLE_LOGGING -DTORRENT_USE_OPENSSL -DTORRENT_USE_LIBCRYPTO -DTORRENT_SSL_PEERS -DOPENSSL_NO_SSL2  -pipe -I/opt/local/libexec/openssl3/include -I/opt/local/libexec/boost/1.81/include -Os -stdlib=libc++ -isysroot/Library/Developer/CommandLineTools/SDKs/MacOSX15.sdk -arch arm64 -MT btfs-btfs.o -MD -MP -MF .deps/btfs-btfs.Tpo -c -o btfs-btfs.o `test -f 'btfs.cc' || echo './'`btfs.cc
:info:build /usr/bin/clang++ -DPACKAGE_NAME=\"btfs\" -DPACKAGE_TARNAME=\"btfs\" -DPACKAGE_VERSION=\"3.1\" -DPACKAGE_STRING=\"btfs\ 3.1\" -DPACKAGE_BUGREPORT=\"johan.gunnarsson@gmail.com\" -DPACKAGE_URL=\"https://github.com/johang/btfs\" -DPACKAGE=\"btfs\" -DVERSION=\"3.1\" -DHAVE_STDIO_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1 -DHAVE_STRINGS_H=1 -DHAVE_SYS_STAT_H=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_UNISTD_H=1 -DSTDC_HEADERS=1 -DHAVE_MEMSET=1 -DHAVE_MEMCPY=1 -DHAVE_STRCMP=1 -DHAVE_MKDIR=1 -DHAVE_STRDUP=1 -DHAVE_REALPATH=1 -DHAVE_GETXATTR=1 -DHAVE_DIRNAME=1 -DHAVE_BASENAME=1 -DHAVE_LIBPTHREAD=1 -I.   -I/opt/local/libexec/openssl3/include -I/opt/local/libexec/boost/1.81/include -isystem/opt/local/include -I/opt/local/include/fuse -isysroot/Library/Developer/CommandLineTools/SDKs/MacOSX15.sdk -std=gnu++14 -Wall -Wextra -Wconversion -Wsign-compare -Wsign-conversion -Wno-unused-parameter -pipe -I/opt/local/libexec/openssl3/include -I/opt/local/libexec/boost/1.81/include -Os -stdlib=libc++ -isysroot/Library/Developer/CommandLineTools/SDKs/MacOSX15.sdk -arch arm64 -MT btfsstat-btfsstat.o -MD -MP -MF .deps/btfsstat-btfsstat.Tpo -c -o btfsstat-btfsstat.o `test -f 'btfsstat.cc' || echo './'`btfsstat.cc
:info:build mv -f .deps/btfsstat-btfsstat.Tpo .deps/btfsstat-btfsstat.Po
:info:build /usr/bin/clang++ -std=gnu++14 -Wall -Wextra -Wconversion -Wsign-compare -Wsign-conversion -Wno-unused-parameter -pipe -I/opt/local/libexec/openssl3/include -I/opt/local/libexec/boost/1.81/include -Os -stdlib=libc++ -isysroot/Library/Developer/CommandLineTools/SDKs/MacOSX15.sdk -arch arm64  -L/opt/local/libexec/openssl3/lib -L/opt/local/libexec/boost/1.81/lib -L/opt/local/lib -Wl,-headerpad_max_install_names -Wl,-syslibroot,/Library/Developer/CommandLineTools/SDKs/MacOSX15.sdk -arch arm64 -o btfsstat btfsstat-btfsstat.o  -lpthread 
:info:build btfs.cc:269:9: warning: 'ec' is deprecated [-Wdeprecated-declarations]
:info:build   269 |         if (a->ec) {
:info:build       |                ^
:info:build /opt/local/include/libtorrent/alert_types.hpp:328:3: note: 'ec' has been explicitly marked deprecated here
:info:build   328 |                 TORRENT_DEPRECATED error_code ec;
:info:build       |                 ^
:info:build /opt/local/include/libtorrent/aux_/deprecated.hpp:37:31: note: expanded from macro 'TORRENT_DEPRECATED'
:info:build    37 | # define TORRENT_DEPRECATED [[deprecated]]
:info:build       |                               ^
:info:build btfs.cc:304:40: warning: 'torrent_added_alert' is deprecated [-Wdeprecated-declarations]
:info:build   304 | handle_torrent_added_alert(libtorrent::torrent_added_alert *a, Log *log) {
:info:build       |                                        ^
:info:build /opt/local/include/libtorrent/alert_types.hpp:247:9: note: 'torrent_added_alert' has been explicitly marked deprecated here
:info:build   247 |         struct TORRENT_DEPRECATED_EXPORT torrent_added_alert final : torrent_alert
:info:build       |                ^
:info:build /opt/local/include/libtorrent/aux_/export.hpp:151:37: note: expanded from macro 'TORRENT_DEPRECATED_EXPORT'
:info:build   151 | #  define TORRENT_DEPRECATED_EXPORT TORRENT_DEPRECATED TORRENT_EXPORT
:info:build       |                                     ^
:info:build /opt/local/include/libtorrent/aux_/deprecated.hpp:37:31: note: expanded from macro 'TORRENT_DEPRECATED'
:info:build    37 | # define TORRENT_DEPRECATED [[deprecated]]
:info:build       |                               ^
:info:build btfs.cc:344:38: warning: 'torrent_added_alert' is deprecated [-Wdeprecated-declarations]
:info:build   344 |         case libtorrent::torrent_added_alert::alert_type:
:info:build       |                                             ^
:info:build /opt/local/include/libtorrent/alert_types.hpp:247:9: note: 'torrent_added_alert' has been explicitly marked deprecated here
:info:build   247 |         struct TORRENT_DEPRECATED_EXPORT torrent_added_alert final : torrent_alert
:info:build       |                ^
:info:build /opt/local/include/libtorrent/aux_/export.hpp:151:37: note: expanded from macro 'TORRENT_DEPRECATED_EXPORT'
:info:build   151 | #  define TORRENT_DEPRECATED_EXPORT TORRENT_DEPRECATED TORRENT_EXPORT
:info:build       |                                     ^
:info:build /opt/local/include/libtorrent/aux_/deprecated.hpp:37:31: note: expanded from macro 'TORRENT_DEPRECATED'
:info:build    37 | # define TORRENT_DEPRECATED [[deprecated]]
:info:build       |                               ^
:info:build btfs.cc:347:17: warning: 'torrent_added_alert' is deprecated [-Wdeprecated-declarations]
:info:build   347 |                         (libtorrent::torrent_added_alert *) a, log);
:info:build       |                                      ^
:info:build /opt/local/include/libtorrent/alert_types.hpp:247:9: note: 'torrent_added_alert' has been explicitly marked deprecated here
:info:build   247 |         struct TORRENT_DEPRECATED_EXPORT torrent_added_alert final : torrent_alert
:info:build       |                ^
:info:build /opt/local/include/libtorrent/aux_/export.hpp:151:37: note: expanded from macro 'TORRENT_DEPRECATED_EXPORT'
:info:build   151 | #  define TORRENT_DEPRECATED_EXPORT TORRENT_DEPRECATED TORRENT_EXPORT
:info:build       |                                     ^
:info:build /opt/local/include/libtorrent/aux_/deprecated.hpp:37:31: note: expanded from macro 'TORRENT_DEPRECATED'
:info:build    37 | # define TORRENT_DEPRECATED [[deprecated]]
:info:build       |                               ^
:info:build btfs.cc:364:30: warning: 'stats_alert' is deprecated [-Wdeprecated-declarations]
:info:build   364 |         case libtorrent::stats_alert::alert_type:
:info:build       |                                     ^
:info:build /opt/local/include/libtorrent/alert_types.hpp:1835:9: note: 'stats_alert' has been explicitly marked deprecated here
:info:build  1835 |         struct TORRENT_DEPRECATED_EXPORT stats_alert final : torrent_alert
:info:build       |                ^
:info:build /opt/local/include/libtorrent/aux_/export.hpp:151:37: note: expanded from macro 'TORRENT_DEPRECATED_EXPORT'
:info:build   151 | #  define TORRENT_DEPRECATED_EXPORT TORRENT_DEPRECATED TORRENT_EXPORT
:info:build       |                                     ^
:info:build /opt/local/include/libtorrent/aux_/deprecated.hpp:37:31: note: expanded from macro 'TORRENT_DEPRECATED'
:info:build    37 | # define TORRENT_DEPRECATED [[deprecated]]
:info:build       |                               ^
:info:build btfs.cc:593:24: warning: 'add_default_plugins' is deprecated [-Wdeprecated-declarations]
:info:build   593 |                 libtorrent::session::add_default_plugins |
:info:build       |                                      ^
:info:build /opt/local/include/libtorrent/session_handle.hpp:805:3: note: 'add_default_plugins' has been explicitly marked deprecated here
:info:build   805 |                 TORRENT_DEPRECATED static constexpr session_flags_t add_default_plugins = 0_bit;
:info:build       |                 ^
:info:build /opt/local/include/libtorrent/aux_/deprecated.hpp:37:31: note: expanded from macro 'TORRENT_DEPRECATED'
:info:build    37 | # define TORRENT_DEPRECATED [[deprecated]]
:info:build       |                               ^
:info:build btfs.cc:594:24: warning: 'start_default_features' is deprecated [-Wdeprecated-declarations]
:info:build   594 |                 libtorrent::session::start_default_features;
:info:build       |                                      ^
:info:build /opt/local/include/libtorrent/session_handle.hpp:811:3: note: 'start_default_features' has been explicitly marked deprecated here
:info:build   811 |                 TORRENT_DEPRECATED static constexpr session_flags_t start_default_features = 1_bit;
:info:build       |                 ^
:info:build /opt/local/include/libtorrent/aux_/deprecated.hpp:37:31: note: expanded from macro 'TORRENT_DEPRECATED'
:info:build    37 | # define TORRENT_DEPRECATED [[deprecated]]
:info:build       |                               ^
:info:build btfs.cc:602:22: warning: 'stats_notification' is deprecated [-Wdeprecated-declarations]
:info:build   602 |                 libtorrent::alert::stats_notification |
:info:build       |                                    ^
:info:build /opt/local/include/libtorrent/alert.hpp:232:3: note: 'stats_notification' has been explicitly marked deprecated here
:info:build   232 |                 TORRENT_DEPRECATED
:info:build       |                 ^
:info:build /opt/local/include/libtorrent/aux_/deprecated.hpp:37:31: note: expanded from macro 'TORRENT_DEPRECATED'
:info:build    37 | # define TORRENT_DEPRECATED [[deprecated]]
:info:build       |                               ^
:info:build btfs.cc:604:22: warning: 'progress_notification' is deprecated [-Wdeprecated-declarations]
:info:build   604 |                 libtorrent::alert::progress_notification |
:info:build       |                                    ^
:info:build /opt/local/include/libtorrent/alert.hpp:220:3: note: 'progress_notification' has been explicitly marked deprecated here
:info:build   220 |                 TORRENT_DEPRECATED
:info:build       |                 ^
:info:build /opt/local/include/libtorrent/aux_/deprecated.hpp:37:31: note: expanded from macro 'TORRENT_DEPRECATED'
:info:build    37 | # define TORRENT_DEPRECATED [[deprecated]]
:info:build       |                               ^
:info:build btfs.cc:676:16: warning: 'session' is deprecated [-Wdeprecated-declarations]
:info:build   676 |         session = new libtorrent::session(pack, flags);
:info:build       |                       ^
:info:build /opt/local/include/libtorrent/session.hpp:208:3: note: 'session' has been explicitly marked deprecated here
:info:build   208 |                 TORRENT_DEPRECATED
:info:build       |                 ^
:info:build /opt/local/include/libtorrent/aux_/deprecated.hpp:37:31: note: expanded from macro 'TORRENT_DEPRECATED'
:info:build    37 | # define TORRENT_DEPRECATED [[deprecated]]
:info:build       |                               ^
:info:build btfs.cc:1007:21: error: assigning to 'typeof(int (*)(const char *, struct fuse_darwin_attr *, struct fuse_file_info *))' (aka 'int (*)(const char *, struct fuse_darwin_attr *, struct fuse_file_info *)') from incompatible type 'int (const char *, struct stat *, struct fuse_file_info *)': type mismatch at 2nd parameter ('struct fuse_darwin_attr *' vs 'struct stat *')
:info:build  1007 |         btfs_ops.getattr = btfs_getattr;
:info:build       |                            ^~~~~~~~~~~~
:info:build btfs.cc:1008:21: error: assigning to 'typeof(int (*)(const char *, void *, fuse_darwin_fill_dir_t, off_t, struct fuse_file_info *, enum fuse_readdir_flags))' (aka 'int (*)(const char *, void *, int (*)(void *, const char *, const fuse_darwin_attr *, long long, fuse_fill_dir_flags), long long, fuse_file_info *, fuse_readdir_flags)') from incompatible type 'int (const char *, void *, fuse_fill_dir_t, off_t, struct fuse_file_info *, enum fuse_readdir_flags)' (aka 'int (const char *, void *, int (*)(void *, const char *, const stat *, long long, fuse_fill_dir_flags), long long, fuse_file_info *, fuse_readdir_flags)'): type mismatch at 3rd parameter ('fuse_darwin_fill_dir_t' (aka 'int (*)(void *, const char *, const fuse_darwin_attr *, long long, fuse_fill_dir_flags)') vs 'fuse_fill_dir_t' (aka 'int (*)(void *, const char *, const stat *, long long, fuse_fill_dir_flags)'))
:info:build  1008 |         btfs_ops.readdir = btfs_readdir;
:info:build       |                            ^~~~~~~~~~~~
:info:build btfs.cc:1011:20: error: assigning to 'typeof(int (*)(const char *, struct statfs *))' (aka 'int (*)(const char *, struct statfs *)') from incompatible type 'int (const char *, struct statvfs *)': type mismatch at 2nd parameter ('struct statfs *' vs 'struct statvfs *')
:info:build  1011 |         btfs_ops.statfs = btfs_statfs;
:info:build       |                           ^~~~~~~~~~~
:info:build 10 warnings and 3 errors generated.
:info:build make[1]: *** [btfs-btfs.o] Error 1
:info:build make[1]: Leaving directory `/opt/local/var/macports/build/_Users_Admin_Developer_Ports_fuse_btfs/btfs/work/btfs-3.1/src'
:info:build make: *** [all-recursive] Error 1
:info:build make: Leaving directory `/opt/local/var/macports/build/_Users_Admin_Developer_Ports_fuse_btfs/btfs/work/btfs-3.1'
```

--- a/src/btfs.cc
+++ b/src/btfs.cc
@@ -19,6 +19,10 @@ along with BTFS.  If not, see <http://www.gnu.org/licenses/>.
 
 #define FUSE_USE_VERSION 31
 
+#ifdef __APPLE__
+#define FUSE_DARWIN_ENABLE_EXTENSIONS 0
+#endif
+
 #include <cstdlib>
 #include <iostream>
 #include <fstream>
@@ -733,15 +737,9 @@ btfs_listxattr(const char *path, char *data, size_t len) {
 	return xattrslen;
 }
 
-#ifdef __APPLE__
-static int
-btfs_getxattr(const char *path, const char *key, char *value, size_t len,
-		uint32_t position) {
-#else
 static int
 btfs_getxattr(const char *path, const char *key, char *value, size_t len) {
 	uint32_t position = 0;
-#endif
 	char xattr[16];
 	int xattrlen = 0;
 

